crypto/x509/v3_ncons.c

```diff
@@ -581,6 +591,54 @@ static int nc_dns(ASN1_IA5STRING *dns, ASN1_IA5STRING *base)
 
 }
 
+/*
+ * This function implements comparison between ASCII/U-label in eml
+ * and A-label in base according to RFC 8398, section 6.
+ * Convert base to U-label and ASCII-parts of domain names, for base
+ * Octet-to-octet comparison of `eml` and `base` hostname parts
+ * (ASCII-parts should be compared in case-insensitive manner)
+ */
+static int nc_email_eai(ASN1_UTF8STRING *eml, ASN1_IA5STRING *base)
+{
+    const char *baseptr = (char *)base->data;
+    const char *emlptr = (char *)eml->data;
+    const char *emlat = strchr(emlptr, '@');
```

Comment:
This should be `strrchr()`. The local part of an email address can contain `@` inside quotes or as a `quoted-pair`. A more strict parser would ensure that what follows the `@` is a valid domain-part, and we're not inside double-quotes in a "local-part only" address with an internal `@`.  Do we need to avoid matching `localpart\@more.local.part` as having `more.local.part` as a domain? Perhaps the existing code for `rfc822Mailbox` has similar concerns.

Reply:
I am not sure we should use ```strrchr()``` because we do not use it in the nc_email function. Current implications presume a "simple" format of the email address and implementing the more universal parser is a definitely out of the scope of this PR.

Reply:
My take is that legacy mistakes do not justify digging the hole deeper.  The domain-part of email address follows the **last** `@` sign in the address, the local-part that precedes it can contain other instances of `@`.  It may even be appropriate to include a commit that fixes this issue in the existing code that handles rfc822Mailbox.

Reply:
Done.